src/fondant/core/component_spec.py

```diff
@@ -61,11 +61,54 @@ class ComponentSpec:
     Class representing a Fondant component specification.
 
     Args:
-        specification: The fondant component specification as a Python dict
+        name: The name of the component
+        image: The docker image uri to use for the component
+        description: The description of the component
+            consumes: A mapping containing the fields consumed by the operation. The keys are the
+            names of the fields to be received by the component, while the values are the
+            type of the field.
+
+
+            produces: A mapping containing the fields produced by the operation. The keys are the
+                names of the fields to be produced by the component, while the values are the
+                type of the field to be written
+
+            arguments: A dictionary containing the argument name and value for the operation.
+
     """
 
-    def __init__(self, specification: t.Dict[str, t.Any]) -> None:
-        self._specification = copy.deepcopy(specification)
+    def __init__(
+        self,
+        name: str,
+        image: str,
+        *,
+        description: t.Optional[str] = None,
+        consumes: t.Optional[t.Dict[str, t.Union[str, pa.DataType]]] = None,
+        produces: t.Optional[t.Dict[str, t.Union[str, pa.DataType]]] = None,
+        arguments: t.Optional[t.Dict[str, t.Any]] = None,
+    ):
+        spec_dict: t.Dict[str, t.Any] = {
+            "name": name,
+            "image": image,
+        }
+
+        if description:
+            spec_dict["description"] = description
+
+        if consumes:
+            spec_dict["consumes"] = consumes
+        else:
+            spec_dict["consumes"] = {"additionalProperties": True}
```

Comment:
I think this logic is specific to the lightweight python components, and doesn't hold in the general case.

I would say that our specification is a [closed-world contract](https://en.wikipedia.org/wiki/Closed-world_assumption) (in contrast to eg. OpenAPI, which is an open-world contract). So if `consumes` is empty, `additionalProperties` should be `False` (the default).

For lightweight python components, `consumes` and `produces` should be `{"additionalProperties": True}` by default, since the user is expected to override them when applying the component.

So I would have the lightweight python components pass in `{"additionalProperties": True}` and don't set any default here.